packages/devtools_app/lib/src/debugger/debugger_model.dart

```diff
@@ -196,42 +243,402 @@ class StackFrameAndSourcePosition {
+  Variable variable, {
+  bool expandAll = false,
+}) async {
+  final ref = variable.ref;
+  if (!variable.isExpandable || variable.treeInitializeStarted || ref == null)
+    return;
+  variable.treeInitializeStarted = true;
+
+  final isolateRef = ref.isolateRef;
+  final instanceRef = ref.instanceRef;
+  final diagnostic = ref.diagnostic;
+  if (diagnostic != null && includeDiagnosticPropertiesInDebugger) {
+    final service = diagnostic.inspectorService;
+    Future<void> _addPropertiesHelper(
+        List<RemoteDiagnosticsNode> properties) async {
+      if (properties == null) return;
+      await addExpandableChildren(
+        variable,
+        await _createVariablesForDiagnostics(
+          service,
+          properties,
+          isolateRef,
+        ),
+        expandAll: true,
+      );
+    }
+
+    if (diagnostic.inlineProperties?.isNotEmpty ?? false) {
+      await _addPropertiesHelper(diagnostic.inlineProperties);
+    } else {
+      assert(!service.disposed);
+      if (!service.disposed) {
+        await _addPropertiesHelper(await diagnostic.getProperties(service));
+      }
+    }
+  }
+  final existingNames = <String>{};
+  for (var child in variable.children) {
+    final name = child?.name;
+    if (name != null && name.isNotEmpty) {
+      existingNames.add(name);
+      if (!isPrivate(name)) {
+        // Assume private and public names with the same name reference the same
+        // data so showing both is not useful.
+        existingNames.add('_$name');
+      }
+    }
+  }
+  if (instanceRef != null) {
+    try {
+      final dynamic result = await serviceManager.service
+          .getObject(variable.ref.isolateRef.id, instanceRef.id);
+      if (result is Instance) {
+        if (result.associations != null) {
+          variable.addAllChildren(
+              _createVariablesForAssociations(result, isolateRef));
+        } else if (result.elements != null) {
+          variable
+              .addAllChildren(_createVariablesForElements(result, isolateRef));
+        } else if (result.bytes != null) {
+          variable.addAllChildren(_createVariablesForBytes(result, isolateRef));
+          // Check fields last, as all instanceRefs may have a non-null fields
+          // with no entries.
+        } else if (result.fields != null) {
+          variable.addAllChildren(_createVariablesForFields(result, isolateRef,
+              existingNames: existingNames));
+        }
+      }
+    } on SentinelException {
+      // Fail gracefully if calling `getObject` throws a SentinelException.
+    }
+  }
+  if (diagnostic != null && includeDiagnosticChildren) {
+    // Always add children last after properties to avoid confusion.
+    final ObjectGroup service = diagnostic.inspectorService;
+    final diagnosticChildren = await diagnostic.children;
+    if (diagnosticChildren?.isNotEmpty ?? false) {
+      final childrenNode =
+          Variable.text(diagnosticChildren.length > 1 ? 'children' : 'child');
```

Comment:
use the pluralize helper from utils.dart for this

Reply:
done